internal/server/run.go

```
@@ -0,0 +1,92 @@
+package server
+
+import (
+	"errors"
+	"io"
+	"log"
+	"net"
+	"tcp-load-balancer/internal/upstream"
+)
+
+var ErrUninitialized = errors.New("load balancer not initialized")
+
+// Run handles incoming connections until terminated.
+func (l *LoadBalancer) Run() error {
+	if l == nil || l.listener == nil {
+		return ErrUninitialized
+	}
+
+	for {
+		// TODO: Set up mTLS in next PR. For now, connect without TLS.
+		clientConn, err := l.listener.Accept()
+		if err != nil {
+			log.Printf("Load balancer unexpectedly declined connection and will be shut down: %s", err)
+			// TODO: attempt to re-establish the listener with a retry mechanism (leaving out of scope for this project).
+			break
```

Why not just return an error  to library used  instead of logging and returning nil in case of `err != nil` ?

---

That would be better. I'll do that.

---

Updated.